tests/daemon/mock-clock.c

```diff
@@ -0,0 +1,138 @@
+#include "mock-clock.h"
+
+typedef enum _MockClocks {
+  MOCK_CLOCK_TYPE_MONOTONIC = 0
+} MockClockType;
+
+struct _MockClock
+{
+  GObject parent_instance;
+
+  /* Microseconds, indexed by MockClockType */
+  gint64 times[1];
+};
+
+typedef struct _MockClockSource {
+  GSource parent;
+
+  MockClock *clock;
+  MockClockType type_;
+
+  /* Microseconds */
+  guint interval;
+  gint64 next_ready_time;
```

Comment:
I borrowed this from eos-payg. In that project, the ready time is split into two 32-bit fields:

https://github.com/endlessm/eos-payg/blob/master/libeos-payg/fake-clock.c#L113-L114

This turns out to be because on 32-bit ARM, `GSource` has 32-bit alignment, and putting a 64-bit variable into that struct (that extends `GSource`) makes it require 64-bit alignment, and the compiler (rightly) complains.

https://github.com/endlessm/eos-payg/commit/3f0144210c6e5aa1e8f24a9a5e8df533e1be6f21

We no longer support 32-bit architectures though.

Reply:
(Ignoring 32-bit makes all this code *way* shorter.)